recipes-core/initrdscripts/files/ptest/test_ni_provisioning.sh

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+#
+# Test that the booted storage disk matches the partition scheme which we
+# expect from the ni_provisioning script. This ptest only supports the efi-ab
+# bootscheme present on new x64, RAUC-based targets.
+exec 2>&1
+
+
+PTEST_TEST='ni_provisioning'
+
+print_error() {
+	echo ERROR: $@ >&2
+	found_error=true
+}
+
+ptest_fail() {
+	echo "FAIL: $PTEST_TEST"
+	exit 1
+}
+
+ptest_pass() {
+	echo "PASS: $PTEST_TEST"
+	exit 0
+}
+
+ptest_skip() {
+	echo "SKIP: $PTEST_TEST"
+	exit 77
+}
+
+
+test_part_id () {
+	local device=${1}
+	local label=${2}
+	local uuid=${3}
+
+	local correct_re_label=${4}
+	local correct_re_uuid=${5}
+
+	echo "test_part_id() ${@:1:3}"
+
+	if [[ ! "${label}" =~ $correct_re_label ]]; then
+		print_error "Partition ${label} @ ${device} does not match required label ${correct_re_label}"
+		found_error=true
+	fi
+	if [[ ! "${uuid}" =~ $correct_re_uuid ]]; then
+		print_error "Partition ${label} @ ${device} does not match required type UUID ${correct_re_uuid}"
+		found_error=true
+	fi
+}
+
+
+found_error=false
+
+if uname --machine | grep -q arm; then
+	echo "INFO: detected an ARM machine architecture." \
+	     "This ptest is only valid for non-ARM, RAUC systems."
+	ptest_skip
+fi
+
+
+## PARTITION TESTS ##
+echo "## Checking that RAUC partitons are present..."
+RE_ESP_UUID=C12A7328\-F81F\-11D2\-BA4B\-00A0C93EC93B
```

Comment:
Please put a comment in here about where/what this is for future reference.  Look at meta-nilrt/recipes-core/initrdscripts/files/disk_config_x64 for reference.

Reply:
Will fix.